scipy/stats/_morestats.py

```diff
@@ -4061,26 +4076,38 @@ def directionalmean(samples, *, axis=0, normalize=True):
 
     .. code-block:: python
 
-        mean=samples.mean(axis=0)
-        directionalmean = mean/np.linalg.norm(mean)
+        mean = samples.mean(axis=0)
+        mean_resultant_length = np.linalg.norm(mean)
+        mean_direction = mean / mean_resultant_length
 
     This definition is appropriate for *directional* data (i.e. vector data
     for which the magnitude of each observation is irrelevant) but not
     for *axial* data (i.e. vector data for which the magnitude and *sign* of
     each observation is irrelevant).
 
+    The function also returns the length of mean resultant vector which
+    can be used to calculate directional variance. Several definitions of
+    directional variance have been proposed e.g. ``Var(z) = 1 - R`` (where
+    ``R`` is the length of the mean resultant vector),
+    ``Var(z) = 1 - R**2`` and ``Var(z) = 2 * (1 - R)``. `directional_stats`
+    instead returns the length of the mean resultant vector so one can
+    choose the appropriate definition based on the use case.
```

Comment:
```suggestion
    Several definitions of directional variance involving the mean resultant
    length ``R`` have been proposed, including ``1 - R`` [1]_, ``1 - R**2`` 
    [2]_, and ``2 * (1 - R)`` [2]_. Rather than choosing one, this function
    returns ``R`` as attribute `mean_resultant_length` so the user can compute
    their preferred measure of dispersion.
```

Reply:
Done.